freqtrade/optimize/space/decimalspace.py

```diff
@@ -1,47 +1,26 @@
-import numpy as np
-from skopt.space import Integer
+from optuna.distributions import FloatDistribution
 
 
-class SKDecimal(Integer):
+class SKDecimal(FloatDistribution):
```

Comment:
@viotemp1 you'll have noticed that i updated this to a optuna replacement.

I've had a few other comments (mostly questions) - which i think would make this mostly a drop-in replacement...

Reply:
Thanks! I've seen that. I'm still working on transforming SKDecimal to optuna, but the class freqtrade.optimize.space.decimalspace.SKDecimal (which is derived from optuna.distributions.BaseDistribution) is passed to optuna and this is not right.  So I'll keep investigating this. 
In order to merge my changes with yours, I should pull again my branch, fix everything and push?

Reply:
thanks, most likely I'll do it tomorrow after running some more tests.

Reply:
Hi, I made all changes and I'm ready to push the changes. The only remaining issue is related to decimal precision which I'm still trying to use in SKDecimal.
This is the test code:
```
from decimal import Decimal, localcontext
import decimal
low = Decimal('0')
high = Decimal('1')
step = 0.001
decimals = 3

d_high = decimal.Decimal(str(high))
d_low = decimal.Decimal(str(low))
d_step = decimal.Decimal(str(step))
with localcontext(prec=decimals) as ctx:
    d_r = d_high - d_low
    print(f"d_high: {d_high}, d_low: {d_low}, d_r: {d_r}, d_step: {d_step}")
    print(d_r % d_step)
```
which gives the error decimal.InvalidOperation: [<class 'decimal.DivisionImpossible'>].
Changing the decimals to 4 or higher works.
Should I drop this with localcontext(prec=decimals) as ctx:?

Reply:
I'm not entirely sure what you're trying to accomplish with this (why is this needed?!?)

In my testing, my approach for SKDecimal works (we accept either "step" or "decimals" - and convert decimals to step).
You simply have to convert 3 to 0.001 - which is simply `1/10**decimals`.

With that, we can use what optuna already supports, without messing with Decimal (i've got very bad experience messing with decimal precision - and personally think, it's a module that should be removed from python :laughing: )

Reply:
ok, decimal removed.